### ISSUE TITLE
Multiple python LSP servers configured but only the first is running

### ISSUE THREAD

**Issue by newtonne** [yegappan/lsp#384]
When multiple LSP servers are configured for a filetype, only the first one seems to get loaded on vim startup.

## Steps to reproduce

Minimal vimrc:

```
let lspServers = [
\   #{
\       name: 'pylsp',
\       filetype: 'python',
\       path: exepath('pylsp'),
\       debug: v:true,
\   },
\   #{
\       name: 'pyright',
\       filetype: 'python',
\       path: exepath('pyright-langserver'),
\       args: ['--stdio'],
\       debug: v:true,
\   },
\]
autocmd VimEnter * call LspAddServer(lspServers)
```

Then install both LSP servers:

```
python3 -m venv venv
. venv/bin/activate
pip install pyright python-lsp-server
```

Then open a python file in vim (`vim test.py`) and run `:LspShowAllServers`:

```
Filetype Information
====================
Filetype: 'python'
Server Name: 'pylsp'
Server Path: '/private/tmp/project/venv/bin/pylsp'
Status: Running

Filetype: 'python'
Server Name: 'pyright'
Server Path: '/private/tmp/project/venv/bin/pyright-langserver'
Status: Running

Buffer Information
==================
Buffer: 'test.py'
Server Path: '/private/tmp/project/venv/bin/pylsp'
Status: Running
```

Only the first configured LSP seems to get loaded. Running `:LspServer restart` fixes it:˚

```
Buffer Information
==================
Buffer: 'test.py'
Server Path: '/private/tmp/project/venv/bin/pylsp'
Status: Running
Server Path: '/private/tmp/project/venv/bin/pyright-langserver'
Status: Running
```

And both LSPs are there when I open another python file (`:e test2.py`):

```
Buffer: 'test2.py'
Server Path: '/private/tmp/project/venv/bin/pylsp'
Status: Running
Server Path: '/private/tmp/project/venv/bin/pyright-langserver'
Status: Running
```

**Comment by maximyurevich** [comment]
Why you use 2 servers for completion?

**Comment by newtonne** [comment]
They don't both get used for completion. See `:h lsp-multiple-servers` for more info on how this can work.

**Comment by girishji** [comment]
I can reproduce this.

@andlrc This is the problematic code in `lsp.vim`. Any reason why this check is in place? I can remove this and submit a PR. Any side effects?

```
export def AddFile(bnr: number): void
  if buf.BufHasLspServer(bnr)
    # LSP server for this buffer is already initialized and running
    return
  endif
```

**Comment by girishji** [comment]
> I'm starting to think that the `pyright-langserver` is silently crashing on it's first setup.

@andlrc that is not true. `AddFile()` gets called twice. First time with only `pylsp` in server list (return value of `LspGetServers()`), and second time with both pylsp and pyright. This is clearly a bug.

**Comment by 00dani** [comment]
There's a simple workaround for this bug, provided you don't need to add new LSP servers to your configuration at runtime. (You probably don't need to do that.) Rather than using an autocommand to add servers after Vim has loaded packages, like this:
```vim
autocmd VimEnter * ++once g:LspAddServer(lspServers)
```
You can intentionally autoload the necessary function ahead of time, like this:
```vim
g:lsp#lsp#AddServer(lspServers)
```
This syntax will configure all servers at launch, before `AddFile()` is ever invoked, so both servers for the same filetype will work correctly. This approach works fine if you installed yegappan/lsp with Vim's packages feature too.

In theory you should also be able to write `import autoload 'lsp/lsp.vim'` and then call `lsp.AddServer(lspServers)`, but if you're using Vim packages then the new `import autoload` syntax [doesn't work that way yet](https://github.com/vim/vim/issues/13313). 🤷‍♀️

**Comment by ts826848** [comment]
Did a bit of digging and came across some peculiar behavior. tl;dr: The number of LSPs returned by `LspGetServers()` seems to depend on whether your LSP configuration is in `after/ftplugin/python.vim` or your main vimrc. Having the configuration in `after` results in 1 of the 2 LSPs being passed, while having the exact same configuration in your main vimrc results in both LSPs being passed.

----

I had the following configuration in my `after/ftplugin/python.vim`:

```vimscript
	let s:lspServers = [
		\#{
			\name: 'pyrefly',
			\filetype: ['python'],
			\path: 'pyrefly',
			\args: ['lsp'],
		\},
		\#{
			\name: 'ruff',
			\filetype: ['python'],
			\path: 'ruff',
			\args: ['server'],
		\},
	\]
	call lsp#lsp#AddServer(s:lspServers)
```

If I add some `echom`s after [this line](https://github.com/yegappan/lsp/blob/39bc35e52d7ab1e66d5d8dcce4012f2cb8afc295/autoload/lsp/lsp.vim#L544), I can see that only the `pyrefly` config is returned.

However, if I move that exact same configuration to my main vimrc, I can see that both the `pyrefly` and `ruff` configs are returned, and both servers show as running when I run `LspShowAllServers`.

At this point I get some more interesting behavior (I don't get semantic highlighting until I restart the servers or unless I swap the positions of the configs), but I figure that's a different bug and so should go in a different issue.
